src/Stratis.Bitcoin.IntegrationTests/EnvironmentMockUpHelpers/FastCoinsDateTimeProvider.cs

```diff
@@ -0,0 +1,63 @@
+﻿using System;
+using Stratis.Bitcoin.Utilities;
+using Stratis.Bitcoin.Utilities.Extensions;
+
+namespace Stratis.Bitcoin.IntegrationTests.EnvironmentMockUpHelpers
+{
+    public sealed class FastCoinsDateTimeProvider : IDateTimeProvider
+    {
+        private TimeSpan adjustedTimeOffset;
+        private DateTime startFrom;
+
+        public FastCoinsDateTimeProvider()
+        {
+            this.adjustedTimeOffset = TimeSpan.Zero;
+            this.startFrom = new DateTime(2018, 1, 1);
+        }
+
+        public long GetTime()
+        {
+            return this.startFrom.ToUnixTimestamp();
+        }
+
+        public DateTime GetUtcNow()
+        {
+            return DateTime.UtcNow;
+        }
+
+        /// <summary>
+        /// This gets called when the Transaction's time gets set.
```

Comment:
I do wonder about summaries like this where it says where something gets called from - on the face of it, it seems useful to know that information, but, without going to the other code, we wouldn't know if it really does or doesn't get called from there.  Therefore a summary like this creates potential work for the reader/maintainer and can easily be inaccurate.   (For reference, In [Clean Code](https://www.amazon.co.uk/Clean-Code-Handbook-Software-Craftsmanship/dp/0132350882), Uncle Bob describes something similar to this in the "Bad comments" section of the book, regarding a properties file getting default values when file not found, yet nothing in the code right there showing if true or not.  Just like this part of the code doesn't show where it is called from, without hunting that down, we cannot know if this is true or not. Furthermore, if that changes later to not call this, would we remember to come and update this comment. It sounds like a huge overhead in code maintenance for such a simple thing, but my gut feel is that saying where something is called from, should solely be in the code alone and not the comments.

Reply:
@majikandy OK yeah I agree with you, I actually put this there for my own reference as this DTP substitutes the actual DTP of the node for this integration test. I needed to keep track of what DTP method was called when, at block creation. I will update the comment to be more specific as to the function of this class.

Reply:
Ah, I missed that this was in tests, I saw a summary and assumed it was prod code